code/modules/antagonists/changeling/changeling.dm

```diff
@@ -74,324 +102,442 @@
 		break
 
 /datum/antagonist/changeling/Destroy()
-	QDEL_NULL(cellular_emporium)
 	QDEL_NULL(emporium_action)
-	. = ..()
-
-/datum/antagonist/changeling/proc/create_actions()
-	cellular_emporium = new(src)
-	emporium_action = new(cellular_emporium)
-	emporium_action.Grant(owner.current)
+	QDEL_NULL(cellular_emporium)
+	return ..()
 
 /datum/antagonist/changeling/on_gain()
-	create_actions()
-	reset_powers()
+	create_emporium()
+	create_innate_actions()
 	create_initial_profile()
 	if(give_objectives)
 		forge_objectives()
 	owner.current.grant_all_languages(FALSE, FALSE, TRUE) //Grants omnitongue. We are able to transform our body after all.
 	owner.current.playsound_local(get_turf(owner.current), 'sound/ambience/antag/ling_aler.ogg', 100, FALSE, pressure_affected = FALSE, use_reverb = FALSE)
-	. = ..()
+	return ..()
+
+/datum/antagonist/changeling/apply_innate_effects(mob/living/mob_override)
+	var/mob/mob_to_tweak = mob_override || owner.current
+	if(!isliving(mob_to_tweak))
+		return
+
+	var/mob/living/living_mob = mob_to_tweak
+	handle_clown_mutation(living_mob, "You have evolved beyond your clownish nature, allowing you to wield weapons without harming yourself.")
+	RegisterSignal(living_mob, COMSIG_MOB_LOGIN, .proc/on_login)
+	RegisterSignal(living_mob, COMSIG_LIVING_LIFE, .proc/on_life)
+	living_mob.hud_used?.lingchemdisplay.invisibility = 0
+	living_mob.hud_used?.lingchemdisplay.maptext = FORMAT_CHEM_CHARGES_TEXT(chem_charges)
+
+	if(!iscarbon(mob_to_tweak))
+		return
+
+	var/mob/living/carbon/carbon_mob = mob_to_tweak
+	RegisterSignal(carbon_mob, list(COMSIG_MOB_MIDDLECLICKON, COMSIG_MOB_ALTCLICKON), .proc/on_click_sting)
```

Comment:
are antagonist datums qdeleted? or are they like gasmixtures where we just free when we're done using them

Reply:
antag datums self-qdel at the end of `/datum/antagonist/proc/on_removal()`